wire-ios-sync-engine/Source/Use cases/ImportBackupUseCase/ImportBackupUseCase.swift

```diff
@@ -164,12 +165,16 @@ struct ImportBackupUseCase: ImportBackupUseCaseProtocol {
             let outputStream = OutputStream(url: decryptedURL, append: false)
         else { throw ImportBackupError.failedToCreateStreamForDecryption }
 
-        try streamDecryptor.decrypt(
-            input: inputStream,
-            output: outputStream,
-            accountID: accountID,
-            password: password
-        )
+        do {
+            try streamDecryptor.decrypt(
+                input: inputStream,
+                output: outputStream,
+                accountID: accountID,
+                password: password
+            )
+        } catch WireCrypto.ChaCha20Poly1305.StreamEncryption.EncryptionError.mismatchingUUID {
```

Comment:
I tried to add a unit test for this, but a linker error prevented me from committing it.